File: lenses-alerts-java/src/main/java/io/lenses/alerting/plugin/javaapi/Alert.java
/*
 * Copyright 2017-2019 Lenses.io Ltd
 */
package io.lenses.alerting.plugin.javaapi;

import java.util.Map;
import java.util.Optional;

final public class Alert implements io.lenses.alerting.plugin.Alert {
    public final Map<String, String> labels;
    public final Map<String, String> annotations;
    public final String startsAt;
    public final Optional<String> endsAt;
    public final String generatorURL;
    public final int alertId;

    public Alert(Map<String, String> labels, Map<String, String> annotations, String startsAt, Optional<String> endsAt,
                 String generatorURL, int alertId) {
        this.labels = labels;
        this.annotations = annotations;
        this.startsAt = startsAt;
        this.endsAt = endsAt;
        this.generatorURL = generatorURL;
        this.alertId = alertId;
    }

    @Override
    public Map<String, String> labels() {
        return labels;
    }

    @Override
    public Map<String, String> annotations() {
        return annotations;
    }

    @Override
    public String startsAt() {
        return startsAt;
    }

    @Override
    public Optional<String> endsAt() {
        return endsAt.empty();
    }

    @Override
    public String generatorURL() {
        return generatorURL;
    }

    @Override
    public int alertId() {
        return alertId;
    }
}

File: lenses-alerts-scala/src/main/scala/io/lenses/alerting/plugin/scalaapi/Alert.scala
/*
 * Copyright 2017-2019 Lenses.io Ltd
 */
package io.lenses.alerting.plugin.scalaapi

import java.util.Optional

import io.lenses.alerting.plugin.{Alert => JAlert}
import io.lenses.alerting.plugin.AlertLevel

import scala.collection.JavaConverters._

case class Alert(level: AlertLevel,
                 category: String,
                 tags: Seq[String],
                 instance: String,
                 summary: String,
                 docs: Option[String],
                 alertId: Int) {
  def asJava: JAlert = {
    new JAlert(level, category, tags.asJava, instance, summary, docs.map(d => Optional.of(d)).getOrElse(Optional.empty[String]()), alertId)
  }
}

object Alert {
  def apply(level: AlertLevel,
            category: String,
            tags: Seq[String],
            instance: String,
            summary: String,
            docs: Option[String],
            alertId: Int): JAlert = {
    new JAlert(level,
      category,
      tags.asJava,
      instance,
      summary,
      docs.map(Optional.of[String]).getOrElse(Optional.empty[String]()),
      alertId
    )
  }
}
